analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

```diff
@@ -0,0 +1,534 @@
+genomic_df %>%
+  group_by(descriptors) %>%
+  tally() %>%
+  arrange(desc(n)) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+```
+
+
+
+```{r}
+genomic_df %>%
+  group_by(Kids_First_Participant_ID, descriptors, experimental_strategy) %>%
+  tally() %>%
+  arrange(desc(n)) %>%
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+
+
+```
+
+
+## Transcriptomic assays
+Looking only at RNA-seq samples and performing the same collapsing of the tumor_descriptor column.
+
+```{r}
+transcriptomic_df <- tumor_descriptor_df %>%
+  filter(experimental_strategy == "RNA-Seq") %>%
+  group_by(Kids_First_Participant_ID) %>%
+  summarize(descriptors = paste(sort(unique(tumor_descriptor)),  
+                                collapse = ", "),
+            experimental_strategy = unique(experimental_strategy),
+            cancer_group = paste(sort(unique(cancer_group)),
+                                          collapse = ", "))
+```
+
+
+```{r}
+transcriptomic_df %>%
+  group_by(descriptors) %>%
+  tally() %>%
+  arrange(desc(n)) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+```
+
+
+
+```{r}
+transcriptomic_df  %>%
+  group_by(Kids_First_Participant_ID, descriptors, experimental_strategy) %>%
+  tally() %>%
+  arrange(desc(n)) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+```
+
+
+## Identify duplicate events - paired assays
+We will identify duplicate events.
+We may want to discard them from further analysis.
+This identifies paired assays as well.
+
+```{r paired_assays_mult_events, echo=TRUE}
+
+# identify multiple events per patient sample
+paired_assays_mult_events <- paired_assays %>%
+  group_by(Kids_First_Participant_ID) %>%
+  summarise(match_id = str_c(match_id, collapse = "~"),
+            experimental_strategy = str_c(experimental_strategy, collapse = "~"),
+            experimental_strategy_sum = str_c(experimental_strategy_sum, collapse = "~")) %>%
+  mutate(has_mult_events = case_when(grepl("~", match_id) ~ "yes", 
+                               TRUE ~ "no"))
+
+table(paired_assays_mult_events$has_mult_events)
+
+# Save paired_assays_mult_events
+# write_tsv(paired_assays_mult_events, path = output_paired_assays_mult_events)
+
+# filter out any multiple events based on has_mult_events column
+# paired_assays_mult_events <- paired_assays_mult_events %>%
+#  filter(has_mult_events == "no")
+
+# table(paired_assays_mult_events$has_mult_events)
+```
+
+
+## Paired genomic, transcriptomic assays
+How many participants have paired genomic and transcriptomic samples for the same tumor_descriptor values?
+
+```{r}
+# if we perform a full join here using the Kids_First_Participant_ID and descriptors, we 
+# will get NAs in columns where pairs don't exist and can use this information 
+# to count
+paired_df <-
+  full_join(genomic_df, transcriptomic_df,
+                   by = c("cancer_group", "Kids_First_Participant_ID", "descriptors", "experimental_strategy"))
+
```

Comment:
I think this is something you can obtain from the `paired_assays` df, once it's modified. Unless there's some difference here that I'm not seeing.